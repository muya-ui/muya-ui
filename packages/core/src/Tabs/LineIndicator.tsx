import styled, { css } from 'styled-components';

import { IThemedBaseProps } from '../types';
import { withThemeForStyled } from '../utils/withTheme';

export interface ILineIndicatorProps {
  $width?: number;
  $left?: number;
  $transition?: 'on' | 'off';
  /**
   * 隐藏指示器
   */
  $hide?: boolean;
}

function tabIndicatorCss(props: ILineIndicatorProps & IThemedBaseProps) {
  const { theme, $left = 0, $width = 0, $transition = 'off', $hide = false } = props;
  const { Tabs: token } = theme.components;
  const { indicator } = token;
  const {
    pattern: { duration, easing },
  } = theme.transition;
  let transitionCss;
  if ($transition === 'on') {
    transitionCss = css`
      transition: all ${duration.status}ms ${easing.status};
    `;
  }
  let indicatorCss;
  if (!$hide) {
    indicatorCss = css`
      z-index: 2;
      &::before {
        content: '';
        display: block;
        position: absolute;
        bottom: 0;
        height: ${indicator.activeHeight}px;
        background: ${indicator.activeColor};
        width: ${$width}px;
        left: ${$left}px;
        ${transitionCss}
      }
    `;
  }
  return css`
    height: ${indicator.bgHeight}px;
    background: ${indicator.bgColor};
    ${indicatorCss}
  `;
}
const LineIndicator = styled.div<ILineIndicatorProps>`
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  ${tabIndicatorCss}
`;

export default withThemeForStyled(LineIndicator);
