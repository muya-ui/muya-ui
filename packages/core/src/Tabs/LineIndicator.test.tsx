import React from 'react';
import renderer from 'react-test-renderer';

import LineIndicator from './LineIndicator';

test('测试 LineIndicator 默认', () => {
  const tree = renderer.create(<LineIndicator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('测试 LineIndicator 正常情况', () => {
  const tree = renderer.create(<LineIndicator $left={0} $width={100} />).toJSON();
  expect(tree).toMatchSnapshot();
});
