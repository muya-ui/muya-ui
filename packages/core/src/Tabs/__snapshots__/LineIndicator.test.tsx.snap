// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 LineIndicator 正常情况 1`] = `
.c0 {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 1px;
  background: rgba(238,238,238,1);
  z-index: 2;
}

.c0::before {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 2px;
  background: rgba(26,122,248,1);
  width: 100px;
  left: 0px;
}

<div
  className="c0"
/>
`;

exports[`测试 LineIndicator 默认 1`] = `
.c0 {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 1px;
  background: rgba(238,238,238,1);
  z-index: 2;
}

.c0::before {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 2px;
  background: rgba(26,122,248,1);
  width: 0px;
  left: 0px;
}

<div
  className="c0"
/>
`;
